hbase-handler/src/test/org/apache/hadoop/hive/hbase/TestHBaseStorageHandler.java

```diff
@@ -46,6 +53,46 @@ public void testHbaseConfigIsAddedToJobConf() {
         jobConfToConfigure.get("hbase.some.fake.option.from.xml.file") != null);
   }
 
+  @Test
+  public void testGetUriForAuth() {
+    try {
+      HBaseStorageHandler hbaseStorageHandler = new HBaseStorageHandler();
+      hbaseStorageHandler.setConf(new JobConf(new HiveConf()));
+      Table table = createMockTable(new HashMap<>());
+      URI uri = hbaseStorageHandler.getURIForAuth(table);
+      // If there is no tablename provided, the default "null" is still
+      // written out. At the time this test was written, this was the current
+      // behavior, so I left this test as/is. Need to research if a null
+      // table can be provided here.
+      Assert.assertEquals("hbase://localhost:2181/null", uri.toString());
+
+      Map<String, String> serdeParams = new HashMap<>();
+      serdeParams.put("hbase.zookeeper.quorum", "testhost");
+      serdeParams.put("hbase.zookeeper.property.clientPort", "8765");
+      table = createMockTable(serdeParams);
+      uri = hbaseStorageHandler.getURIForAuth(table);
+      Assert.assertEquals("hbase://testhost:8765/null", uri.toString());
+
+      serdeParams.put("hbase.table.name", "mytbl");
+      table = createMockTable(serdeParams);
+      uri = hbaseStorageHandler.getURIForAuth(table);
+      Assert.assertEquals("hbase://testhost:8765/mytbl", uri.toString());
+
+      serdeParams.put("hbase.columns.mapping", "mycolumns");
+      table = createMockTable(serdeParams);
+      uri = hbaseStorageHandler.getURIForAuth(table);
+      Assert.assertEquals("hbase://testhost:8765/mytbl/mycolumns", uri.toString());
+
+      serdeParams.put("hbase.table.name", "my#tbl");
+      serdeParams.put("hbase.columns.mapping", "myco#lumns");
+      table = createMockTable(serdeParams);
+      uri = hbaseStorageHandler.getURIForAuth(table);
+      Assert.assertEquals("hbase://testhost:8765/my%23tbl/myco%23lumns", uri.toString());
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
The catch and rethrow pattern is rarely necessary in unit tests. Moreover, it has few disadvantages such as obscuring the original exception & error message, increasing indentation etc. The try-catch block can be removed and the `URISyntaxException` can go into the declaration of the method if necessary.